### ISSUE TITLE
Possible explanation slower early steps

### ISSUE THREAD

**Issue by MarktHart** [KellerJordan/modded-nanogpt#19]
Re: https://github.com/KellerJordan/modded-nanogpt/blob/e1343685254e498c3a069c4d8b6a3ab8aec0a18c/train_gpt2.py#L464C7-L464C90

I assume it's due to torch.compile not being AOT.

Frankly, I would argue that this should be included in the total training time. Reason being that changing the size of the attention window or batch size in the middle of training does require a recompile. Stopping the timer for each recompile opens the door to a whole lot of untimed "this is not gradient descent"-type calculations.

Anyways, do what feels right for you. I really like this repo and the idea behind it.

**Comment by KellerJordan** [comment]
The reason I'm not interested in including compilation time in the final timing is as follows:

1) It varies a lot - the very first run on a new machine will have longer compilation time than subsequent runs

2) In theory it can be completely gotten rid of. For example, Karpathy's llm.c has virtually zero compilation time. So, including compilation it in the timing would probably just force the run to be rewritten in CUDA, which isn't something I'm interested in

You're right that this is a problem for anything which would need an intermediate recompile. Well, I'd be interested in taking that on a case by case basis

Thanks -Keller
